lldb/source/DataFormatters/ValueObjectPrinter.cpp

```diff
@@ -18,39 +18,35 @@
 using namespace lldb;
 using namespace lldb_private;
 
-ValueObjectPrinter::ValueObjectPrinter(ValueObject *valobj, Stream *s) {
-  if (valobj) {
-    DumpValueObjectOptions options(*valobj);
-    Init(valobj, s, options, m_options.m_max_ptr_depth, 0, nullptr);
-  } else {
-    DumpValueObjectOptions options;
-    Init(valobj, s, options, m_options.m_max_ptr_depth, 0, nullptr);
-  }
+ValueObjectPrinter::ValueObjectPrinter(ValueObject &valobj, Stream *s)
```

Comment:
Would it make sense to pass in a "ValueObjectSP" here instead? Could some of our crashes be due to someone creating a ValueObjectPrinter and then somehow the ValueObject gets freed behind out backs?

Reply:
For this series of crashes,  the EXC_BAD_ACCESS address is the correct offset for the ivar being accessed.  So the address of the ValueObject pointer has to be 0x0.  If the pointer had gotten freed, the printer's pointer value wouldn't change, so the access would just be some random looking address.

Also, see the more general comment about uses of the ValueObjectPrinter.